waiter/src/waiter/scheduler.clj

```diff
@@ -47,6 +47,16 @@
   [& args]
   `(log/log "Scheduler" :debug nil (print-str ~@args)))
 
+(defmacro log-service-instance
+  "Log InstanceTracker-specific messages."
+  [instance event-type]
+  `(log/log "InstanceTracker" :debug nil (utils/clj->json (assoc ~instance :timestamp (t/now) :event-type ~event-type))))
+
+(defn swap-atom-with-instance-logging
```

Comment:
This function name should end with `!` (per Clojure conventions) since it mutates an atom.

Reply:
Also, we should be specific about the fields we want to render rather than rendering the entire instance map. E.g. the request log does this where it extracts specific fields from the request/response.

Reply:
We should delegate to `add-instance-to-buffered-collection!` here. I think us *not* using that function in kubernetes.clj was a bug that I introduced.

Reply:
I would prefer our helper function have a simpler name, like `track-failed-instance!`